File: 506-RelativeRanks/main.py
class Solution:
    def findRelativeRanks(self, nums):
        """
        :type nums: List[int]
        :rtype: List[str]
        """
        d = {value: idx for idx, value in enumerate(sorted(nums, reverse=True))}
        
        txt = ["Gold Medal", "Silver Medal", "Bronze Medal"]
        for i, n in enumerate(nums):
            nums[i] = txt[d[n]] if d[n] < 3 else str(d[n]+1)
        
        return nums

File: 506-RelativeRanks/C++/main.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& nums) {
        unordered_map<int, int> table;
        vector<int> copy(nums.begin(), nums.end());
        sort(nums.rbegin(), nums.rend());
        
        for(size_t i = 0; i < nums.size(); ++i) {
            table[nums[i]] = i;
        }
        
        vector<string> ranks{"Gold Medal", "Silver Medal", "Bronze Medal"};
        vector<string> res;
        
        for(size_t i = 0; i < copy.size(); ++i) {
            int n = copy[i];
            res.push_back( table[n] < 3 ? ranks[table[n]] : to_string(table[n] + 1) );
        }
        
        return res;
    }
};